lib/core/CPatternSet.cc

```diff
@@ -26,6 +26,42 @@ CPatternSet::CPatternSet()
       m_ContainsPatterns() {
 }
 
+bool CPatternSet::initFromPatternList(const TStrVec& patterns) {
+    TStrVec fullPatterns;
+    TStrVec prefixPatterns;
+    TStrVec suffixPatterns;
+    TStrVec containsPatterns;
+
+    for (const auto& pattern : patterns) {
+        std::size_t length = pattern.length();
+        if (length == 0) {
+            continue;
+        }
+        if (pattern[0] == WILDCARD) {
+            if (length > 2 && pattern[length - 1] == WILDCARD) {
+                std::string middle = pattern.substr(1, length - 2);
+                containsPatterns.push_back(middle);
+            } else if (length > 1) {
+                std::string suffix = pattern.substr(1);
+                suffixPatterns.push_back(std::string(suffix.rbegin(), suffix.rend()));
+            }
+        } else if (length > 1 && pattern[length - 1] == WILDCARD) {
+            prefixPatterns.push_back(pattern.substr(0, length - 1));
```

Comment:
It's probably not necessary to optimize for performance in this code, but (apart from the one that reverses order) instead of creating extra copies using `substr` you could use `emplace_back` with the `std::string` constructor that copies a substring of another string.